libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+
+    /* ALPN options for AWS IoT. */
+    /* ppcALPNProtos is unused. putting here to align behavior in IotNetworkAfr_Create. */
+    /* coverity[misra_c_2012_rule_4_6_violation] */
+    /* coverity[misra_c_2012_rule_8_13_violation] */
+    const char * ppcALPNProtos[] = { socketsAWS_IOT_ALPN_MQTT };
+
+    /* Set secured option. */
+    TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                0,
+                                                SOCKETS_SO_REQUIRE_TLS,
+                                                NULL,
+                                                0 );
+
+    if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+    {
+        LogError( ( "Failed to set secured option for soocket." ) );
+    }
+
+    /* Set ALPN option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pAlpnProtos != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_ALPN_PROTOCOLS,
+                                                    ppcALPNProtos,
+                                                    sizeof( ppcALPNProtos ) / sizeof( ppcALPNProtos[ 0 ] ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set ALPN option socket." ) );
+        }
+    }
+
+    /* Set SNI option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->disableSni == false ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_SERVER_NAME_INDICATION,
+                                                    pHostName,
+                                                    hostnameLength + 1 );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set SNI option for socket. TransportSocketStatus %d, hostnameLength  + 1 %d ",
+                        TransportSocketStatus, hostnameLength + 1 ) );
+        }
+    }
+
+    /* Set custom server certificate. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pRootCa != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_TRUSTED_SERVER_CERTIFICATE,
+                                                    pSocketsConfig->pRootCa,
+                                                    pSocketsConfig->rootCaSize );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set server certificate option for socket." ) );
+        }
+    }
```

Comment:
Should the client certificate and private key values be configured in the socket as well?

Reply:
we don't need the clients cert/key in the SocketConfig_t anymore, removing it in b719c48142e222a8b58ca00e4d06027d03c6e5da